docs/spec-files/message_structure.md

```diff
@@ -26,12 +26,14 @@ A **DIDComm signed message** is a signed [JWM (JSON Web Messages)](https://tools
 
 **DIDComm signed messages** are not necessary to provide message integrity (tamper evidence), or to prove the sender to the recipient. Both of these guarantees automatically occur with the authenticated encryption in **DIDComm encrypted messages**. **DIDComm signed messages** are only necessary when the origin of plaintext has to be provable to third parties, or when the sender can't be proven to the recipient by authenticated encryption because the recipient is not known in advance (e.g., in a broadcast scenario). Adding a signature when one is not needed [can degrade rather than enhance security because it relinquishes the sender's ability to speak off the record](https://github.com/hyperledger/aries-rfcs/blob/master/concepts/0049-repudiation/README.md#summary). We therefore expect **DIDComm signed messages** to be used in a few cases, but not as a matter of course.
 
-When a message is *both* signed and encrypted, the plaintext is signed, and then the signed envelope is encrypted. The opposite order is not used, since it would imply that the signer committed to opaque data (which is unsafe and undermines non-repudiation).
+When a message is *both* signed and encrypted, the plaintext SHOULD be signed, and then the signed envelope SHOULD be encrypted. The opposite order SHOULD NOT be used, since it would imply that the signer committed to opaque data (which is unsafe and undermines non-repudiation).
```

Comment:
```suggestion
When a message is *both* signed and encrypted, the plaintext MUST be signed, and then the signed envelope MUST be encrypted. The opposite order MUST NOT be used, since it would imply that the signer committed to opaque data (which is unsafe and undermines non-repudiation).
```

I see no justifiable reason to do the opposite ordering, so I'd prefer to cut that code path out with a MUST here.

Reply:
I could see protocols where you have a signature on a ciphertext, in order to allow anyone to verify the provenance of the ciphertext without allowing decryption...

Typically you could do Sign then Encrypt then Mac/Sign... Having a strong requirement here seemed a bit restrictive. 

Also anything that is based on homomorphic encryption might want to not embed the signature within its ciphertext... Arguably, the structure of the plaintext wouldn't allow for such schemes with DIDComm, so it's probably not a problem to rule out such possibilities.

Reply:
I don't see value in adding that additional code path here. I'd prefer to leave these problems out of scope here since it goes against the [recommendations of JOSE](https://datatracker.ietf.org/doc/html/rfc7519#section-11.2). I'd prefer to remain in alignment with this rather than support these additional aspects being suggested here.

Reply:
How about not using normative terminology and linking to that JOSE section?

Reply:
@kdenhartog ?